src/core/thread/csl_tx_scheduler.cpp

```diff
@@ -188,7 +188,8 @@ otError CslTxScheduler::HandleFrameRequest(Mac::TxFrame &aFrame)
         aFrame.SetIsARetransmission(false);
     }
 
-    aFrame.SetChannel(mCslTxChild->GetCslChannel());
+    aFrame.SetChannel(mCslTxChild->GetCslChannel() == 0 ? Get<Mac::Mac>().GetPanChannel()
```

Comment:
Instead of executing this condition statement per frame to send, for efficiency, how about setting the child's CSL channel to be pan channel directly?

Reply:
I think this would make the logic more complex: When the parent changes channel, it might need to change `csl channel` for each of its child. And it doesn't know if the `csl channel` has been specified by the user of CSL receiver.(since we have set child's CSL channel to be a non-zero value).

Reply:
make sense.